internal/events/sql.go

```diff
@@ -27,11 +28,22 @@ const (
 	);`
 	dropEventsTable = `DROP TABLE IF EXISTS ` + schemaName + `.events;`
 
-	insertEventIdempotent  = `INSERT INTO ` + schemaName + `.events (id, data, event_type) VALUES ($1, $2, $3) ON CONFLICT DO NOTHING;`
-	deleteEvent            = `DELETE FROM ` + schemaName + `.events WHERE id = $1;`
-	getEvents              = `SELECT data, event_type FROM ` + schemaName + `.events;`
-	getUnbroadcastedEvents = `SELECT data, event_type FROM ` + schemaName + `.events WHERE NOT received AND NOT broadcasted;`
-	markReceived           = `UPDATE ` + schemaName + `.events SET received = TRUE WHERE id = $1;`
+	insertEventIdempotent = `INSERT INTO ` + schemaName + `.events (id, data, event_type) VALUES ($1, $2, $3) ON CONFLICT DO NOTHING;`
+	deleteEvent           = `DELETE FROM ` + schemaName + `.events WHERE id = $1;`
+
+	deleteEvents = `DELETE FROM ` + schemaName + `.events WHERE id =ANY($1);`
+	getEvents    = `SELECT data, event_type FROM ` + schemaName + `.events;`
+
+	getNewEvents = `SELECT e.data, e.event_type
+	FROM ` + schemaName + `.events AS e
+	LEFT JOIN ` + voting.VotingSchemaName + `.resolutions AS r ON e.id = r.id
+	WHERE r.id IS NULL;`
+
+	// FilterObservedEvents returns the list of events that are observed by the node and not yet broadcasted.
+	filterObservedEvents = `SELECT unnested.id
+	FROM unnest($1::BYTEA[]) AS unnested(id)
+	LEFT JOIN ` + schemaName + `.events AS res ON unnested.id = res.id
+	WHERE res.id IS NOT NULL AND res.broadcasted = FALSE;`
```

Comment:
The `WHERE res.id IS NOT NULL` clause is useless. `res.id` is the primary key, it cannot be null

Reply:
Also, you want to use an inner join here right? Otherwise you are going to be broadcasting vote ids for events you have never observed